src/libraries/System.Private.Xml.Linq/src/System/Xml/Linq/XElement.cs

```diff
@@ -265,6 +267,7 @@ public XName Name
         {
             get
             {
+                Debug.Assert(name != null);
```

Comment:
why do we need to assert here? Should this be nullable with `[DisallowNull]`?

Reply:
I'm just double-checking that `name` is never null even when there are a few internal ctors that does not initialize it.

Reply:
I think is not needed, will remove it.